slashing/program/src/duplicate_block_proof.rs

```diff
@@ -30,6 +48,32 @@ impl SlashingProofData for DuplicateBlockProofData {
         let shred2 = Shred::new_from_payload(self.shred2)?;
         check_shreds(slot, &shred1, &shred2)
     }
+
+    fn unpack(data: &'a [u8]) -> Result<Self, SlashingError>
+    where
+        Self: Sized,
+    {
+        if data.len() < LENGTH_SIZE {
+            return Err(SlashingError::ProofBufferTooSmall);
+        }
+        let (length1, data) = data.split_at(LENGTH_SIZE);
+        let shred1_length = try_from_bytes::<PodU32>(length1)
+            .map_err(|_| SlashingError::ProofBufferDeserializationError)?;
+        let (shred1, data) = data.split_at(u32::from(*shred1_length) as usize);
```

Comment:
To avoid a panic, can this return an error if `data.len() < shred1_length`?

Reply:
Yep good call c018720853875cf9ef88675c68f210733738a775